### DISCUSSION TITLE
URL is not a constructor error

### DISCUSSION THREAD

**Discussion by midori0507** [vercel/next.js#15965] [Help]
Hi,

I've tried to upgrade Next from 9.4.4 to latest and this wild error appear.
This is very strange cause i'm using Node v12.

The strange thing is i try to debug to find reason why this error appear. But it's very randomly happen. Once it happened, all view will have this error when tried to visit.

<img width="963" alt="Screenshot 2020-08-07 at 9 02 20" src="https://user-images.githubusercontent.com/9389531/89614341-e22f6780-d88c-11ea-81f8-68fe6b2c2950.png">

**Comment by timneutkens** [comment]
What browser are you using?

**Reply by midori0507** [reply]
Latest chrome MacOS

**Reply by midori0507** [reply]
Any idea @timneutkens.

Unfortunately i can not reproduce this. But i found similar issue when they tried to replace all require('url') with URL and got this error.

Check this out:
https://github.com/ConsenSys/armlet/issues/105

**Reply by Musacoli** [reply]
I am also facing the same issue, and haven't found a fix for it, @midori0507 , were you able to solve this ?

**Reply by midori0507** [reply]
> I am also facing the same issue, and haven't found a fix for it, @midori0507 , were you able to solve this ?

Yes. Its because the global namespace is polluted by one of our dependency. Its RecordRTC lib, override the URL global

**Reply by midori0507** [reply]
> I am also facing the same issue, and haven't found a fix for it, @midori0507 , were you able to solve this ?

This is not because of Nextjs. Its very tricky to debug which library is overriding the global

**Reply by Musacoli** [reply]
Yes same here, I am trying to use RecordRTC.

**Reply by midori0507** [reply]
Hey, you can checkout my PR on RecordRTC and use my fork as your dependency while waiting for author to fix the issue

**Reply by Musacoli** [reply]
> Hey, you can checkout my PR on RecordRTC and use my fork as your dependency while waiting for author to fix the issue

Thanks a bunch, let me give that a shot.

**Reply by HnM97** [reply]
Thanks!! It really helped a lot !

**Reply by merte143** [reply]
had the same issue. if you don't need RecordRTC on the server, you can use next's dynamic import and only load it on the client when you need e.g.

`const handleRecording = async () => {
    const RecordRTC = (await import("recordrtc")).default; ...`

**Reply by catosaurusrex2003** [reply]
[https://github.com/muaz-khan/RecordRTC/issues/795#issuecomment-1138760844](https://github.com/muaz-khan/RecordRTC/issues/795#issuecomment-1138760844)
for RecordRTC this worked best for me.

**Comment by javiermendozain** [comment]
This work for me:

Change this: 
` var URL = window.URL;`

By this:
```
function URL(url, base) {
  this.url = url;
  this.base = base;
  return window.URL;
}
```

on file `node_modules/recordrtc/RecordRTC.js`

I did with that script on package 
``` "postinstall": "cp patch/RecordRTC.js node_modules/recordrtc/RecordRTC.js"```

**Reply by HunterJoey** [reply]
This did work in recordRTC, could you please fork and open a pull request to benifit more people?

**Reply by UltiRequiem** [reply]
I also have problems with `recordRTC` but this didn't work for me 😭 

Any repository where I could see this working?

**Reply by mlnchk** [reply]
@UltiRequiem i've also had problems with recordRTC, solved by [patch-package](https://www.npmjs.com/package/patch-package)

**Reply by kartik-raj7** [reply]
Thanks Man this works
